### DISCUSSION TITLE
Custom image deprecation

### DISCUSSION THREAD

**Discussion by stefanes** [actions/virtual-environments#6310] [General questions] [answered]
Hi,

We have some custom images based on the Ubuntu 18.04 image in this repo. Is there anyway for us to configure the agent to display a warning to all builds using a specific image, like you do on hosted agents?

![2022-09-27 10_05_21-Clipboard](https://user-images.githubusercontent.com/5484354/192470120-e011f91d-99ae-4e2b-933f-04af56a1fcfc.png)

I have tried adding a warning to `/agent/.setup_info` but I am unable to get it generate a proper warning like the one above:

![image](https://user-images.githubusercontent.com/5484354/192470931-4ff125d8-9835-4ee0-ba3f-0c7a9091ed25.png)

**Comment by al-cheb** [comment, marked as answer]
Hey @stefanes.
You could use annotations to add warnings for pipelines - https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-a-warning-message

```
runs-on: ubuntu-20.04
    steps:
      - run: |
          echo "::warning title=deprecation::test warning
```

![image](https://user-images.githubusercontent.com/47745270/192564450-f7ecffed-01bc-4b21-9f31-3c5772a2b373.png)

**Reply by stefanes** [reply]
Sure, but this is about auto-generating a warning in a build, any build, using a specific VM image - just like when you now build on the hosted Ubuntu 18.04 image. So this is outside of any pipeline step where I can use GitHub or Azure Pipelines annotation/logging commands.

So how do you get the deprecation warning to show up in all 18.04 builds?

**Reply by miketimofeev** [reply]
@stefanes hosted pools are managed by service that allows applying such a notifications so it's not possible to do it the same way for self-hosted, I'm afraid

**Reply by stefanes** [reply]
Ok, thanks for your response!

**Reply by Mikebeem** [reply]
You could create a [pipeline decorator](https://learn.microsoft.com/en-us/azure/devops/extend/develop/add-pipeline-decorator?view=azure-devops ) to add the warning to all pipelines using specific agents. I don't think you can use the pool name as conditional insertion, but maybe the name of your agents is sufficient to filter on?
